core/ui/PageTemplate/modals.tid

```diff
@@ -0,0 +1,93 @@
+list-after: 
+tags: $:/tags/PageTemplate
+title: $:/core/ui/PageTemplate/modals
+
+\whitespace trim
+
+\define close-current-modal()
+<$action-deletetiddler $tiddler=<<stateTiddler>> />
+\end
+
+\define tm-close-handler()
+<!-- Close the modal unless a different tiddler was given in the message parameter -->
+<$list filter="[<event-param>!is[blank]!match<currentTiddler>]" emptyMessage=<<close-current-modal>> >
+<$action-sendmessage $message="tm-close-tiddler" $param=<<event-param>>/>
+</$list>
+</$let>
+\end
+
```

Comment:
IMO Instead of using `<$let stateTiddler` inside the macro you could define a function outside the macro

```
\function stateTiddler() [removeprefix[$:/state/modal/]sortan[]add[1]else[0]addprefix[$:/state/modal/]]
```